prune.go

```diff
@@ -129,122 +135,194 @@ func txnsPruneC(txnsName string) string {
 // with a bit of luck something like this will one day be part of
 // mgo/txn.
 func PruneTxns(db *mgo.Database, txns *mgo.Collection) error {
-	present := struct{}{}
-
-	// Load the ids of all completed txns and all collections
-	// referred to by those txns.
-	//
-	// This set could potentially contain many entries, however even
-	// 500,000 entries requires only ~44MB of memory. Given that the
-	// memory hit is short-lived this is probably acceptable.
-	txnIds := make(map[bson.ObjectId]struct{})
-	collNames := make(map[string]struct{})
+	workingSetName := txns.Name + ".prunetemp"
```

Comment:
I take it, it was easier to project into another collection in the same DB rather than a second DB?